kclvm/sema/src/advanced_resolver/mod.rs

```diff
@@ -65,13 +67,14 @@ pub struct AdvancedResolver<'ctx> {
 
 pub struct Context<'ctx> {
     pub program: &'ctx Program,
-    node_ty_map: IndexMap<AstIndex, TypeRef>,
+    node_ty_map: IndexMap<AstIndex, Arc<Type>>,
```

Comment:
Should we consider modifying all `TypeRef` definitions from Rc<Type>to Arc<Type>, and what are the reasons for making this modification?

Reply:
lsp needs to communicate between threads, Rc<Type> is unsafe

Reply:
https://github.com/kcl-lang/kcl/pull/827